ghost/core/test/e2e-api/admin/members.test.js

```diff
@@ -1384,6 +1388,126 @@ describe('Members API', function () {
         should.deepEqual(memberWithPaidSubscription, readMember, 'Editing a member returns a different format than reading a member');
     });
 
+    it('Cannot add complimentary subscriptions to a member with an active subscription', async function () {
+        if (!memberWithPaidSubscription) {
+            // Previous test failed
+            this.skip();
```

Comment:
This is an interesting pattern, have you seen this used elsewhere in the tests?

I'm not against it, as it is frustrating when a single test failing causes a cascade of later ones failing.

Ultimately that's because we have tests depending on some global state outside of the test, which IMO is not good.

We _could_ fix this by making sure each test does the setup that it needs, but that adds overhead.

Using this pattern at least makes the coupling explicit 🤔 Though what would happen if we removed the previous test, it would render this test useless right?

I'd be interested in your thoughts either way on this, and if you think it's a good idea - let's talk with Hannah about sharing/using this pattern going forwards

Reply:
Yeah, I introduced this pattern a couple of months ago in the same file. I didn't really think a lot about it when writing these additional tests. I don't think this pattern is ideal, but it was an effort to prevent having to repeat all the Stripe mocking in every test. But moving that repeated part to the top might be a better solution... 🤔 Or we could build the models manually without mocking Stripe...

Reply:
Gonna move this to a tech debt issue and clean up the whole test file in one go 🧹

Reply:
@SimonBackx what was the outcome of the cleanup? If the `memberWithPaidSubscription` is used in a group of test, like it seems to be done here, probably best would be grouping such tests into a "describe" and having a `memberWithPaidSubscription` be set up through fixtures? 🤔